compiler/src/dotty/tools/dotc/ast/Desugar.scala

```diff
@@ -1726,12 +1726,26 @@ object desugar {
         }
         else {
           // Desugar [T_1, ..., T_M] -> (x_1: P_1, ..., x_N: P_N) => body
-          // Into    new scala.PolyFunction { def apply[T_1, ..., T_M](x_1: P_1, ..., x_N: P_N) = body }
+          // with pt [S_1, ..., S_M] -> (O_1, ..., O_N) => R
+          // Into    new scala.PolyFunction { def apply[T_1, ..., T_M](x_1: P_1, ..., x_N: P_N): R2 = body }
+          // where R2 is R, with all references to S_1..S_M replaced with T1..T_M.
```

Comment:
What if `R` contains term references to `O_i`?

Reply:
I'm not sure that's possible.  Is it?  I tried `[O] => ((u: Unit), PingMessage[O]) => (u.type, O)` and `[O] => (u, PingMessage[O]) => (u.type, O)`, but nothing.